File: app/src/main/java/life/plank/juna/zone/data/network/model/FixtureByDate.java
package life.plank.juna.zone.data.network.model;

import java.util.Date;
import java.util.List;

import lombok.Data;

@Data
public class FixtureByDate {
    private Date date;
    private List<MatchFixture> fixtures;

    private FixtureByDate(Date date, List<MatchFixture> fixtures) {
        this.date = date;
        this.fixtures = fixtures;
    }

    public static FixtureByDate getFrom(Date date, List<MatchFixture> matchFixtureList) {
        return new FixtureByDate(date, matchFixtureList);
    }
}

File: app/src/main/java/life/plank/juna/zone/data/model/football/FixtureByDate.kt
package life.plank.juna.zone.data.model.football

import android.os.Parcelable
import kotlinx.android.parcel.*
import java.util.*
import kotlin.collections.ArrayList

@Parcelize
data class FixtureByDate(
        var date: Date = Date(),
        var fixtures: @RawValue MutableList<MatchFixture> = Collections.emptyList()
) : Parcelable

fun FixtureByDate.put(date: Date, fixture: MatchFixture): FixtureByDate {
    this.date = date
    this.fixtures = ArrayList()
    fixtures.add(fixture)
    return this
}